src/org/thoughtcrime/securesms/translation/TranslationTask.java

```diff
@@ -0,0 +1,63 @@
+package org.thoughtcrime.securesms.translation;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.thoughtcrime.securesms.database.model.MessageRecord;
+import org.thoughtcrime.securesms.util.JsonUtils;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class TranslationTask extends AsyncTask<String, Void, TranslationResult> {
+    private String API_KEY;
+    private OkHttpClient httpClient;
+    private MessageRecord messageRecord;
+    public TranslationResultHandler resultHandler;
+
+    public TranslationTask(MessageRecord messageRecord) {
+        this.httpClient = new OkHttpClient();
+        this.messageRecord = messageRecord;
+    }
+
+    @Override
+    protected TranslationResult doInBackground(String... strings) {
+        TranslationResult result = null;
+        try {
+            result = getTranslationResult(strings[0], strings[1]);
+        } catch (IOException e) {
+            Log.e("Translation", "IOException");
+        }
+        return result;
+    }
+
+    @Override
+    protected void onPostExecute(TranslationResult result) {
+        resultHandler.processTranslationResult(result, messageRecord);
+    }
+
+    public TranslationTask(OkHttpClient client, MessageRecord messageRecord) {
```

Comment:
It would read easier to have the two constructors together at the top.

Reply:
Agreed